syntax="proto3";

message getProduct{
    string id = 1;
}

message setProduct{
    string id = 1;
    int32 quantity=2;
}

message addToOrder{
    string id = 1;
    int32 quantity =2;
}

message confirmOrder{
    string id = 1;
}

message TransactionResponse{
    int32 status_code = 1;
}

message NullMessage{}

message Package {
    string address=1;
    int32 productQuantity=2;
}

message ShipmentConfirmation{
    string aproximateDate =2;
}

service Inventory{
    rpc Ready(NullMessage) returns (NullMessage);
    rpc setProducts(setProduct) returns (TransactionResponse);
    rpc getProducts(getProduct) returns (TransactionResponse);
}

service ShoppingCart{
    rpc AddProduct(addToOrder) returns (TransactionResponse);
    rpc ConfirmOrder(confirmOrder) returns (TransactionResponse);
}

service Shipment{
    rpc ShipmentRegister(Package) returns (ShipmentConfirmation);
}

